lib/spack/spack/cmd/include.py

```diff
@@ -0,0 +1,162 @@
+
+description = "manage included configs and concrete environments in the active environment"
+section = "environments"
+level = "long"
+
+subcommands = ["add", "remove", "list"]
+
+
+def include_list_setup_parser(subparser):
+    """List include(s) in the active environment"""
+
+    expand_group = subparser.add_mutually_exclusive_group()
+    expand_group.add_argument("--expand", action="store_true")
+    expand_group.add_argument("--no-expand", action="store_false", dest="expand")
+
+    # TODO: Add recursive listing option for environments including environments with include
+    #       The default for list should be to only list the comonents listed in the active
+    #       environment.
+
+
+def include_list(env, args):
+    include = env.manifest.list_includes(concrete=args.concrete)
+
+    if sys.stdout.isatty():
+        if include:
+            tty.msg(
+                "Included "
+                + ("Concrete Environments" if args.concrete else "Configuration Scopes")
+            )
+        else:
+            tty.msg("No includes found")
+
+    if include:
+        msg = ""
+        for inc in include:
+            if args.expand:
+                formatted_inc = substitute_path_variables(inc)
+            else:
+                # Collapse the prefix to the active environment or spack root variables
+                formatted_inc = inc.replace(env.path, "$env")
+                formatted_inc = formatted_inc.replace(spack.paths.spack_root, "$spack")
+
+            # Print the env name and path
+            env_name = ev.environment_name(inc)
+            if not env_name == inc:
+                formatted_inc = f"{env_name} ({formatted_inc})"
+
+            msg += f"\n\t{formatted_inc}"
+
+        tty.msg(msg)
+
+
+def include_add_setup_parser(subparser):
+    """Add include(s) to the active environment"""
+    subparser.add_argument(
+        "includes",
+        metavar="includes",
+        nargs="+",
+        help="List of path(s) to configs or environments to include",
+    )
+
+
+def include_add(env, args):
+    with env.write_transaction():
+        invalid = env.manifest.add_includes(args.includes, concrete=args.concrete)
+        env.write()
+
+        if not len(invalid) == len(args.includes):
+            msg = f"Adding includes to {env.name}"
+            for inc in args.includes:
+                if inc not in invalid:
+                    msg += f"\n\t{inc}"
+            tty.msg(msg)
+
+        if invalid:
+            msg = "Invalid Paths"
+            for p in invalid:
+                msg = msg + f"\n\t{p}"
+
+            tty.warn(msg)
+
+
+def include_remove_setup_parser(subparser):
+    """Remove include(s) from the active environment"""
+    subparser.add_argument(
+        "includes",
+        metavar="includes",
+        nargs="+",
+        help="List of path(s) to configs or environments to include",
+    )
+
+
+def include_remove(env, args):
+    with env.write_transaction():
```

Comment:
Would be nice to be able to support the name of a Spack-managed environment here as well.